test/relations.test.js

```diff
@@ -3629,6 +4828,7 @@ describe('relations', function () {
           });
           cb(null, summary);
         });
+        return cb.promise;
       };
 
       summarize.shared = true; // remoting
```

Comment:
A general note on writing promise-based code that applies to most of the tests in this pull request: one of the benefits of using promises is that you can keep your code nesting shallow by "lifting" an intermediary promise from a `then` block to the higher level.

```js
Category.findOne()
  .then(function(cat) {
    return cat.jobs.summarize();
  });
  .then(function(summary) {
    // ...
    done();
  })
  .catch(done);
```

Instead of deeply nested

```js
Category.findOne()
  .then(function(cat) {
    return cat.jobs.summarize()
      .then(function(summary) {
        // ...
       done();
      });
  })
  .catch(done);
```

Notice that the first variant reads more like a traditional sequential program, while the second looks more like a callback hell/xmas tree.

While it would be nice to use the first style thorough this pull request, I understand such change may take quite a long time and I can live with the current version too (I won't insist on fixing the coding style).